src/Utilities/Compiler/WellKnownTypeProvider.cs

```diff
@@ -44,51 +35,6 @@ static WellKnownTypeProvider CreateWellKnownTypeProvider(Compilation compilation
 
         public Compilation Compilation { get; }
 
-        /// <summary>
```

Comment:
All the properties here are the ones that are very commonly accessed in dataflow analysis. Given that this well known type provider is not specific to flow analysis anymore, it makes sense to remove them from here. However, I think we should still move all of these to [DataflowOperationVisitor](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs). They could be implemented in this file as following:

```csharp
protected INamedTypeSymbol Exception { get; } = WellKnownTypeProvider.GetOrCreateTypeByMetadataName(SystemException);
```

Reply:
I can also do this as a follow-up change if you want to avoid another iteration.

Reply:
That's fine, I am taking care to add one more commit for this change. Thank you for the feedback.